### PR TITLE
Open room member avatar in a media viewer

### LINKED ISSUE #1907: Open room members avatars in a media viewer
When the avatar of a room member is clicked in the member details screen, we should be able to open it in an image viewer.

### PR DESCRIPTION
**Pull request by jmartinesp**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

- Moves `MediaViewer` associated code to `:libraries:mediaviewer`, from `:features:messages`.
- Reuses this viewer to display the room member avatar.

## Motivation and context

Closes #1907.

## Tests

- Get into a room, select a user so its details screen is displayed.
- Tap on the avatar in this screen, it should open a media viewer.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 11

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/model/event/TimelineItemAudioContent.kt [hunk 1]
```diff
@@ -29,6 +29,10 @@ data class TimelineItemAudioContent(
     val fileExtension: String,
 ) : TimelineItemEventContent {
 
-    val fileExtensionAndSize = formatFileExtensionAndSize(fileExtension, formattedFileSize)
+    val fileExtensionAndSize =
+        io.element.android.libraries.mediaviewer.api.helper.formatFileExtensionAndSize(
```
Comment (bmarty) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/model/event/TimelineItemAudioContent.kt:33]: Maybe cleanup?

File: features/messages/impl/src/test/kotlin/io/element/android/features/messages/impl/timeline/factories/event/TimelineItemContentMessageFactoryTest.kt [hunk 2]
```diff
@@ -496,7 +496,7 @@ class TimelineItemContentMessageFactoryTest {
         featureFlagService: FeatureFlagService = FakeFeatureFlagService()
     ) = TimelineItemContentMessageFactory(
         fileSizeFormatter = FakeFileSizeFormatter(),
-        fileExtensionExtractor = FileExtensionExtractorWithoutValidation(),
+        fileExtensionExtractor = io.element.android.libraries.mediaviewer.api.util.FileExtensionExtractorWithoutValidation(),
```
Comment (bmarty) [inline features/messages/impl/src/test/kotlin/io/element/android/features/messages/impl/timeline/factories/event/TimelineItemContentMessageFactoryTest.kt:499]: Same

File: features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsFlowNode.kt [hunk 3]
```diff
@@ -136,9 +148,31 @@ class RoomDetailsFlowNode @AssistedInject constructor(
...
+                    mediaInfo = MediaInfo(
+                        name = navTarget.userName,
+                        mimeType = mimeType,
+                        formattedFileSize = "",
+                        fileExtension = ""
+                    ),
+                    mediaSource = MediaSource(url = navTarget.avatarUrl),
+                    thumbnailSource = MediaSource(url = navTarget.avatarUrl),
+                    canDownload = false,
+                    canShare = false,
```
Comment (bmarty) [inline features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsFlowNode.kt:172]: Why restricting sharing and downloading here?

File: features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/details/avatar/RoomMemberAvatarPreviewNode.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import dagger.assisted.AssistedInject
+import io.element.android.anvilannotations.ContributesNode
+import io.element.android.libraries.di.RoomScope
+
+@ContributesNode(RoomScope::class)
+class RoomMemberAvatarPreviewNode @AssistedInject constructor(
+    @Assisted buildContext: BuildContext,
+    @Assisted plugins: List<Plugin>,
+    presenterFactory: io.element.android.libraries.mediaviewer.api.viewer.MediaViewerPresenter.Factory,
+) : io.element.android.libraries.mediaviewer.api.viewer.MediaViewerNode(buildContext, plugins, presenterFactory)
```
Comment (bmarty) [inline features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/details/avatar/RoomMemberAvatarPreviewNode.kt:31]: clean up?

**Review by bmarty — APPROVED** [review]
Thanks for doing this, maybe do some cleanup before merging?

**Comment by jmartinesp** [comment]
Thanks for finding these fully qualified usages. I moved lots of classes around so AS broke the imports, and it seems like I only found and fixed some of those.

**Pull request merged**